### PR TITLE
Implement full dealer, transaction, and item logic with comprehensive…

### PR DESCRIPTION
**Pull request by yarchick-balaban**
… tests

- Implemented models and relationships for User, Dealer, Transaction, and Item.
- Added business logic: users can create dealers, transactions, and assign items to transactions with individual prices.
- Ensured each transaction is linked to a dealer; added support for stub dealers for non-purchased items.

- Added and refactored tests:
  - Dealer and transaction creation.
  - Adding items to existing transactions.
  - Creating items with new transactions and dealers.
  - Enforcing that transactions cannot exist without a dealer.
  - Automatic creation of stub dealer and transaction for found items.

- All key business rules and edge-cases are now covered by the test suite.

### CODE REVIEW

File: backend/src/models/dealer.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+if TYPE_CHECKING:
+    from .user import User
+    from .transaction import Transaction
+
+
+class Dealer(Base):
+    __tablename__ = 'dealers'
+    id = Column(Integer, primary_key=True)
+    name = Column(String(255), nullable=False)
```
Comment (topol-3000) [inline backend/src/models/dealer.py:15]: name: Mapped[str] = mapped_column(String(255))

File: backend/src/models/dealer.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+if TYPE_CHECKING:
+    from .user import User
+    from .transaction import Transaction
+
+
+class Dealer(Base):
+    __tablename__ = 'dealers'
+    id = Column(Integer, primary_key=True)
```
Comment (topol-3000) [inline backend/src/models/dealer.py:14]: id is extra, just use IdIntPkMixin 
`class Dealer(Base, IdIntPkMixin):`

File: backend/src/models/dealer.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    name = Column(String(255), nullable=False)
+    contact_info = Column(String(255), nullable=True)
```
Comment (topol-3000) [inline backend/src/models/dealer.py:16]: the same as above

File: backend/src/models/dealer.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
```
Comment (topol-3000) [inline backend/src/models/dealer.py:17]: do we need direct relation to user?

File: backend/src/models/transaction.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+if TYPE_CHECKING:
+    from .dealer import Dealer
+    from .user import User
+    from .item import Item
+
+
+class Transaction(Base):
+    __tablename__ = 'transactions'
+    id = Column(Integer, primary_key=True)
```
Comment (topol-3000) [inline backend/src/models/transaction.py:15]: IdIntPkMixin

File: backend/src/models/transaction.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    dealer_id = Column(Integer, ForeignKey('dealers.id'), nullable=False)
+    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
```
Comment (topol-3000) [inline backend/src/models/transaction.py:17]: Mapped and mapped_column

File: backend/src/models/user.py [hunk 7]
```diff
@@ -20,6 +21,8 @@ class User(Base, IdIntPkMixin, SQLAlchemyBaseUserTable[UserIdType]):
     # Relationships
     items: Mapped[list['Item']] = relationship('Item', back_populates='user', lazy='select')
     collections: Mapped[list['Collection']] = relationship('Collection', back_populates='user', lazy='select')
+    # Relationship to dealers (категории)
+    dealers: Mapped[list['Dealer']] = relationship('Dealer', back_populates='user', lazy='select')
```
Comment (topol-3000) [inline backend/src/models/user.py:25]: why do we need dealers directly related to user?
may be user-transaction and transaction-dealer is enough?

File: backend/src/schemas/dealer.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
+from pydantic import BaseModel, Field
+from typing import Optional
+from schemas.base import SchemaConfigMixin
+
+
+class DealerBase(SchemaConfigMixin):
+    name: str = Field(..., min_length=1, max_length=255)
```
Comment (topol-3000) [inline backend/src/schemas/dealer.py:7]: name: Annotated[str, Field(min_length=1, max_length=255, description="some description...')

File: backend/src/schemas/item.py [hunk 9]
```diff
@@ -16,7 +16,8 @@ class ItemBase(SchemaConfigMixin):
 
 
 class ItemCreate(ItemBase):
-    pass
+    transaction_id: Optional[int] = None
+    price: Optional[float] = None
```
Comment (topol-3000) [inline backend/src/schemas/item.py:20]: price is extra at all or move it to the ItemBase

Comment (topol-3000) [reply backend/src/schemas/item.py:20]: now it looks like we should just create Item with price field, but we will read all fields without transaction_id and price

File: backend/src/schemas/transaction.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+from pydantic import BaseModel, Field
+from typing import Optional, List
+from datetime import date
+from .item import ItemRead
+
+
+class TransactionBase(BaseModel):
+    dealer_id: int
```
Comment (topol-3000) [inline backend/src/schemas/transaction.py:8]: Annotated[...]

File: backend/src/tests/conftest.py [hunk 11]
```diff
@@ -40,7 +40,7 @@ async def test_db_engine():
 @pytest_asyncio.fixture(scope="function")
 async def test_session(test_db_engine) -> AsyncGenerator[AsyncSession, None]:
     """Create a test database session."""
-    async with AsyncSession(test_db_engine) as session:
+    async with AsyncSession(test_db_engine, expire_on_commit=False) as session:
```
Comment (topol-3000) [inline backend/src/tests/conftest.py:43]: I'm not sure it's a good way to use expire_on_commit=False in tests

File: backend/src/api/routes/transactions.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+@router.get('/', response_model=list[TransactionRead])
+async def get_transactions(
+    session: SessionDependency,
+    current_user: User = Depends(current_active_user),
+):
+    result = await session.execute(
+        select(Transaction).where(Transaction.user_id == current_user.id)
+    )
+    return result.scalars().all()
```
Comment (topol-3000) [inline backend/src/api/routes/transactions.py:43]: What about update and delete?
we have to add at least delete method in case of user created transaction by mistake

File: backend/src/api/routes/dealers.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+    current_user: User = Depends(current_active_user),
+):
+    result = await session.execute(
+        select(Dealer).where(Dealer.id == dealer_id, Dealer.user_id == current_user.id)
+    )
+    dealer = result.scalar_one_or_none()
+    if not dealer:
+        raise HTTPException(status_code=404, detail='Dealer not found')
+    result = await session.execute(
+        select(Transaction)
```
[the remark this answers is not in the document]
Comment (topol-3000) [reply backend/src/api/routes/dealers.py:99]: And also please check N+1 problem and eager loading.
you should have strong knowlages about this shit to deliver features in the correct way.

I don't always aware of use cases to suggest implementing lazy loading and eager loading. and where we need to solve N+1 problem

File: backend/src/models/dealer.py (hunk 4 above)
Comment (yarchick-balaban) [reply backend/src/models/dealer.py:17]: Yes,  need to use private list of dealers for each user.

File: backend/src/schemas/transaction_item.py [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    Schema for creating a TransactionItem.
+
+    Attributes:
+        item_id (str): Item ID.
+        price (float): Price for this item in transaction.
+        quantity (int): Quantity of this item in transaction.
+    """
+    item_id: Annotated[str, Field(description="Item ID")]
+    price: Annotated[float, Field(description="Price for this item in transaction")]
+    quantity: Annotated[int, Field(description="Quantity of this item in transaction")]
```
Comment (topol-3000) [inline backend/src/schemas/transaction_item.py:32]: do we need the quantity field? we discussed that each item shall be as a separate element in a DB
